native/src/render/nova_renderer.cpp

```diff
@@ -494,6 +500,15 @@ namespace nova {
 
     void nova_renderer::end_frame() {
         NOVA_PROFILER_FLUSH_TO_FILE("profiler_data.txt");
+        std::vector<uint32_t> *buffer = new std::vector<uint32_t>(passes_list.size() * 2);
+        context->device.getQueryPoolResults(context->timestamp_query_pool, 0, buffer->size(), buffer->size() *
```

Comment:
Shouldn't you be using `timestampValidBits` here to see if we need 32-bit or 64-bit values?